src/org/omegat/util/HttpConnectionUtils.java

```diff
@@ -141,23 +141,24 @@ public static boolean downloadBinaryFile(final URL jarFileURL, final Map<String,
         int responseCode = httpURLConnection.getResponseCode();
         if (responseCode == HttpURLConnection.HTTP_OK) {
             String contentType = httpURLConnection.getContentType();
-            int contentLength = httpURLConnection.getContentLength();
-            if (contentType.equals("binary/octet-stream") || contentType.equals("application/jar-archive")) {
+            long contentLength = httpURLConnection.getContentLength();
+            if (contentType.equals("application/octet-stream") || contentType.equals("application/jar-archive")) {
                 try (InputStream inputStream = httpURLConnection.getInputStream();
                      FileOutputStream outputStream = new FileOutputStream(saveFilePath)) {
-                    int bytesRead = -1;
-                    int readLength = 0;
-                    byte[] buffer = new byte[BUFFER_SIZE];
-                    while ((bytesRead = inputStream.read(buffer)) != -1) {
-                        outputStream.write(buffer, 0, bytesRead);
-                    }
+                    long transferred = IOUtils.copy(inputStream, outputStream, BUFFER_SIZE);
+                    assert(transferred == contentLength);
```

Comment:
What is your intent for this assert?

Asserts don't run unless you explicitly enable them, and we don't enable them, so this won't do anything in a release build.

Is that OK? Did you want to raise an exception, or log something instead?

Reply:
Oh, this is python way. I'd like to raise exception

Reply:
Sounds good. FYI see [here](https://docs.oracle.com/javase/7/docs/technotes/guides/language/assert.html#enable-disable) for details on Java assertions.